runtime/drivers/clickhouse/information_schema.go

```diff
@@ -18,46 +19,67 @@ func (c *connection) InformationSchema() drivers.InformationSchema {
 }
 
 func (i informationSchema) All(ctx context.Context) ([]*drivers.Table, error) {
-	q := `
+	conn, release, err := i.c.acquireMetaConn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = release() }()
+
+	var databases []string
+	dbs, ok := i.c.config["databases"].(string)
+	if ok {
+		databases = strings.Split(dbs, ",")
+	} else {
+		row := conn.QueryRowxContext(ctx, "SELECT currentDatabase()")
+		var db string
+		if err := row.Scan(&db); err != nil {
+			return nil, err
+		}
+		databases = append(databases, db)
+	}
+
+	var res []*drivers.Table
+	for _, database := range databases {
+		q := `
 		SELECT
 			T.table_catalog AS DATABASE,
-			T.table_schema AS SCHEMA,
 			T.table_name AS NAME,
 			T.table_type AS TABLE_TYPE, 
 			C.column_name AS COLUMNS,
 			C.data_type AS COLUMN_TYPE,
 			C.ordinal_position as ORDINAL_POSITION
 		FROM information_schema.tables T 
 		JOIN information_schema.columns C ON T.table_schema = C.table_schema AND T.table_name = C.table_name
-		WHERE T.table_schema = currentDatabase()
-		ORDER BY DATABASE, SCHEMA, NAME, TABLE_TYPE, ORDINAL_POSITION
+		WHERE T.table_schema = ?
+		ORDER BY DATABASE, NAME, TABLE_TYPE, ORDINAL_POSITION
 	`
 
-	conn, release, err := i.c.acquireMetaConn(ctx)
-	if err != nil {
-		return nil, err
-	}
-	defer func() { _ = release() }()
+		rows, err := conn.QueryxContext(ctx, q, database)
+		if err != nil {
+			return nil, err
+		}
 
-	rows, err := conn.QueryxContext(ctx, q)
-	if err != nil {
-		return nil, err
-	}
-	defer rows.Close()
+		tables, err := i.scanTables(rows)
+		if err != nil {
+			rows.Close()
+			return nil, err
+		}
 
-	tables, err := i.scanTables(rows)
-	if err != nil {
-		return nil, err
+		rows.Close()
+		res = append(res, tables...)
 	}
-
-	return tables, nil
+	return res, nil
 }
 
-func (i informationSchema) Lookup(ctx context.Context, name string) (*drivers.Table, error) {
-	q := `
+func (i informationSchema) Lookup(ctx context.Context, db, schema, name string) (*drivers.Table, error) {
+	var q string
+	var args []any
+	// table_catalog and table_schema both means the name of the database in which the table is located in clickhouse.
+	// we use either db or schema arg to set table_schema
+	if db == "" && schema == "" {
```

Comment:
It's weird that ClickHouse returns the same for both, but in our conceptual model, arguably these are just "schemas", not "databases". To prevent them showing up with duplicate prefixes "foo_schema.foo_schema.table_name" in the UI, should we maybe return `database` as the empty string always? If `database` is empty (or indeed, if `schema` is empty), our analytical APIs should be disregard them in the implementation.

Reply:
In clickhouse's terminology they are `database`. For example the SQL is `SHOW databases`, `current_database()` and `SHOW schemas` and `current_schema()`.  
> To prevent them showing up with duplicate prefixes

Yeah I was just returning the `database` field and empty `schema` field in information_schema calls. May be I can stop reading `schema` field completely in this lookup call ?

Reply:
Okay maybe I'm missing something. Usually, you can run queries like `select * from table` (uses default DB and schema), `select * from schema.table` (uses default DB), and `select * from db.schema.table`. Are you saying in Clickhouse, you can only do `select * from table` (uses default db/schema) and `select * from db.db.table`?

Because if you can do `select * from db.table` in ClickHouse, then it sounds like that's actually a schema in our terminology (and SQL terminology), and I would argue we should then map the "database" in ClickHouse to the `schema` field in our generic output.

Reply:
Yes we can do `select * from db.table` and `select * from table` and not `select * from db.db.table`. Sure we can map it to schema.